### ISSUE TITLE
Enabling master lock using direct paths causes content to disappear

### ISSUE THREAD

**Issue by lhupitr** [jellyfin/jellyfin-kodi#169]
Using Jellyfin for Kodi with direct paths, initial ingestion completes normally, content is available and can be played but if master lock is enabled in Kodi (even when choosing to not lock down anything) all content disappears. Normally, when using Kodi without Jellyfin, this wouldn't cause content to disappear.

 - OS: Linux OSMC
 - Jellyfin Version: 10.4.3
 - Kodi Version: 18.5
 - Addon Version: 0.40

**Comment by TrueTechy** [comment]
I'm using Ubuntu 18 so there's a difference there but otherwise it's the same and the content does not disappear when enabling the master lock. Could you try disabling the lock and seeing if the content returns?

**Comment by lhupitr** [comment]
Yes, I should have mentioned, disabling the lock causes the content to return. The same thing also happens on an x86-64 pc with arch linux.

**Comment by TrueTechy** [comment]
Do you have the addon set to Native mode? 

Seem this might be related to an issue with Kodi with how content is added by the addon, but this only seems to affect Native mode
https://forum.kodi.tv/showthread.php?tid=208996

**Comment by lhupitr** [comment]
Yes, this occurs in native (direct paths) mode, this does not happen when using the default playback mode

**Comment by TrueTechy** [comment]
We'd need to dig into the differences between how the addon adds to Kodi in the different modes. Do you have a specific need for native mode just now?

**Comment by lhupitr** [comment]
Yes there are various kodi in play addons that I use.

**Comment by mcarlton00** [comment]
We've received reports via Matrix that this is happening with multiple profiles in addon mode as well.

**Comment by MagnificentMagician** [comment]
Same issue here with addon-mode enabled. This bug really annoys me, as having a master lock is mandatory for enforcing age restrictions on movies. The system is an OSMC Vero 4K. I've deleted the .kodi folder, tried the stable OSMC Kodi Leia & beta OSMC Kodi Matrix. After setting a master lock the libraries are empty. There are approx. hundred lines logspam of the following after updating libraries in Jellyfin manually (with master lock enabled):

2021-05-17 19:31:13.090 T:2797     INFO <general>: JELLYFIN.helper.utils -> DEBUG::jellyfin_kodi/helper/utils.py:48 --[ window clear: Jellyfin.nodes.recentepisodes.path ]
2021-05-17 19:31:13.093 T:2797    DEBUG <CAddonSettings[plugin.video.jellyfin]>: trying to load setting definitions from old format...
2021-05-17 19:31:13.097 T:2797    DEBUG <CSettingsManager>: requested setting (LastIncrementalSync) was not found.
2021-05-17 19:31:13.097 T:2797    DEBUG <CSettingsManager>: requested setting (SyncInstallRunDone) was not found.
2021-05-17 19:31:13.097 T:2797    DEBUG <CSettingsManager>: requested setting (groupedSets) was not found.
2021-05-17 19:31:13.097 T:2797    DEBUG <CSettingsManager>: requested setting (platformDetected) was not found.

However, I was able to get a "fix" for this behaviour. The Jellyfin server is running in the same local network as the Kodi client. Using a mobile hotspot for my client solves the issue immediately.

@mcarlton00

**Comment by mcarlton00** [comment]
Running Kodi from a mobile hotspot allowed you to enable a master lock and everything worked properly?

**Comment by MagnificentMagician** [comment]
Yesterday it did but today in a different network it was no longer working. However, when browsing the movie database with Kodi "master mode" enabled it did work in every network. Tested it multiple times. Looks like the Jellyfin for Kodi Addon writes the movies in a protected area, which is only available to users with "master mode" enabled, as far as I see it.

**Comment by Zelgadis87** [comment]
Can confirm that master lock causes issues even in addon mode on Kodi Matrix (not sure about other modes or versions).
Unfortunately there is no evidence of this in the logs, even when in Debug mode. Libraries were empty when synced to Kodi while using master lock (dynamic folders where working perfectly fine), I randomly tried to disable master lock after a few hours of tries, and now all issues are solved.
I can upload logs if needed but there is no error reported in there. Please let me know if I can be of help.

**Comment by mmBesar** [comment]
I have the same problem!

**Comment by Coprocessor8087** [comment]
I too confirm this issue in add-on mode.

**Comment by SteveMavic** [comment]
Same on Kodi 19.4.0 and Jellyfin 10.8.8

**Comment by matmielke** [comment]
Is there any solution planned or at least a workaround available?
I'd like to implement the following scenario:

- One Kodi installation on SmartTV
- One central Jellyfin installation with separate users (family and adults)
- Kodi should have one profile for alle the movies (jellyfin's adult profile) with PIN protection and one family profile without PIN

As soon as I enable the PIN, the content disappears at the moment.

**Comment by samdoshi** [comment]
This works for me:

Add the following to `sources.xml` in `//sources/video`

```xml
    <source>
            <name>Jellyfin</name>
            <path pathversion="1">plugin://plugin.video.jellyfin/</path>
            <allowsharing>true</allowsharing>
    </source>
```
[source](https://forum.kodi.tv/showthread.php?tid=372240&page=2)

If you read through the whole thread it might give you other pointers too.

**Comment by matmielke** [comment]
Great, I gave it a quick try on a test installation and it did the job.
I just had to integrate that into the sources.xml of the respective profile and that's it.
Thank you very much!

**Comment by Hansie9999** [comment]
A workaround was found by a Kodi forum user a while back for me.
[Kodi Forum](https://forum.kodi.tv/showthread.php?tid=372240)
I opened a issue for a permanent fix on the Kodi Github
[Github Kodi](https://github.com/xbmc/xbmc/issues/24358)

I hope a more permanent fix can be implemented in a new update,

**Comment by prahal** [comment]
@Hansie9999 in https://github.com/xbmc/xbmc/issues/24358#issuecomment-2094089265you tell that this bug is fixed in kodi 21 which is stable for a while now (but it seems from the backlog that it was not backported to Kodi 20 Nexus as the issue implies it was not in 20.5 and there was no later release of Nexus as far as I know). 
Shouldn't this jellyfin-kodi also be closed then?
